plugins/statuslines/lualine.nix

```diff
@@ -0,0 +1,129 @@
+{ pkgs, config, lib, ... }:
+with lib;
+let
+  cfg = config.programs.nixvim.plugins.lualine;
+  helpers = import ../helpers.nix { lib = lib; };
+  separators = mkOption {
+    type = types.submodule {
+      options = {
+        left = mkOption {
+          default = " ";
+          type = types.str;
+          description = "left separator";
+        };
+        right = mkOption {
+          default = " ";
+          type = types.str;
+          description = "right separator";
+        };
+      };
+    };
+    default = { };
+  };
+  component_options = mode:
+    mkOption {
+      type = types.submodule {
+        options = {
+          mode = mkOption {
+            type = types.str;
+            default = "${mode}";
+          };
+          icons_enabled = mkOption {
+            type = types.enum [ "True" "False" ];
+            default = "True";
+            description = "displays icons in alongside component";
+          };
+          icon = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = "displays icon in front of the component";
+          };
+          separator = separators;
+        };
+      };
+      default = { };
+    };
+in {
+  options = {
+    programs.nixvim.plugins.lualine = {
+      enable = mkEnableOption "Enable airline";
+
+      options = mkOption {
+        type = types.submodule {
+          options = {
+            theme = mkOption {
+              default = "auto";
+              type = types.str;
+              description = "The theme to use for lualine-nvim.";
+            };
+            section_separators = separators;
+            component_separators = separators;
+            disabled_filestypes = mkOption {
+              type = types.listOf types.str;
+              default = [ ];
+              example = ''[ "lua" ]'';
+              description = "filetypes to disable lualine on";
+            };
+            always_divide_middle = mkOption {
+              type = types.bool;
+              default = true;
+              description =
+                "When true left_sections (a,b,c) can't take over entire statusline";
+            };
+          };
+        };
+        default = { };
+        description = "Options for lualine";
+      };
+      sections = mkOption {
+        type = types.nullOr (types.submodule ({ ... }: {
+          options = {
+            lualine_a = component_options "mode";
```

Comment:
As for these, I think we can just make them a,b,c etc., and drop the lualine_ prefix